lib/shopify_app/middleware/jwt_middleware.rb

```diff
@@ -0,0 +1,41 @@
+module ShopifyApp
+  class JWTMiddleware
+    TOKEN_REGEX = /^Bearer\s+(.*?)$/
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      return call_next(env) unless authorization_header?(env)
+
+      token = extract_token(env)
+      return call_next(env) unless token
+
+      set_env_variables(token, env)
+      call_next(env)
+    end
+
+    private
+
+    def call_next(env)
+      @app.call(env)
+    end
+
+    def authorization_header?(env)
+      env['HTTP_AUTHORIZATION']
+    end
+
+    def extract_token(env)
+      match = env['HTTP_AUTHORIZATION'].match(TOKEN_REGEX)
```

Comment:
If you changed `authorization_header?` to `authorization_header`, you could re-use it here...

Also, you could use that gnarly "one weird trick" I mentioned above to do this in one line:

```ruby
      authorization_header.match(TOKEN_REGEX)&.[](1)
```

(from the category of bad ideas that are so bad they're good?)